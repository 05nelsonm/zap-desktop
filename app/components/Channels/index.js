export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
