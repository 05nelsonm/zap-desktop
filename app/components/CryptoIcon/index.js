import CryptoIcon from './CryptoIcon'

export default CryptoIcon
