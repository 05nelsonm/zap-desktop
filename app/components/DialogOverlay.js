import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
