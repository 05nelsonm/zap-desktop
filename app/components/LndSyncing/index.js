import LndSyncing from './LndSyncing'

export default LndSyncing
