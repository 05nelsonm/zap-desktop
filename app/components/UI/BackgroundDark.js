import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundDark
 * @example
 * <BackgroundDark />
 */
class BackgroundDark extends React.Component {
  render() {
    return <Box bg="darkestBackground" color="primaryText" {...this.props} />
  }
}

export default BackgroundDark
