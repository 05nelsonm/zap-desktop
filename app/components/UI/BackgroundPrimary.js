import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundPrimary
 * @example
 * <BackgroundPrimary />
 */
class BackgroundPrimary extends React.Component {
  render() {
    return <Box bg="primaryColor" color="primaryText" {...this.props} />
  }
}

export default BackgroundPrimary
