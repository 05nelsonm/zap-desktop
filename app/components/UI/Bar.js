import styled from 'styled-components'
import { variant, space, height } from 'styled-system'

const bars = variant({
  key: 'bars',
})

const Bar = styled.hr`
  height: 1px;
  margin: 0;
  padding: 0;
  border: 0;
  background-color: ${props => props.theme.colors.primaryText};
  ${space}
  ${height}
  ${bars}
`

Bar.defaultProps = {
  variant: 'normal',
}

Bar.displayName = 'Bar'

export default Bar
