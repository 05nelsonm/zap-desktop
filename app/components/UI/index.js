export ActionBar from './ActionBar'
export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export Button from './Button'
export Checkbox from './Checkbox'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export Notification from './Notification'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Value from './Value'
export Wizard from './Wizard'
