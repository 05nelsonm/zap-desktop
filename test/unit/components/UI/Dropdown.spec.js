import React from 'react'
import { Dropdown } from 'components/UI'
import renderer from 'react-test-renderer'
import { dark } from 'themes'

const currencies = [
  {
    key: 'btc',
    name: 'BTC',
  },
  {
    key: 'bits',
    name: 'bits',
  },
  {
    key: 'sats',
    name: 'satoshis',
  },
]

const setCurrency = jest.fn()

describe('component.Dropdown', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Dropdown activeKey="btc" items={currencies} onClick={setCurrency} theme={dark} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
