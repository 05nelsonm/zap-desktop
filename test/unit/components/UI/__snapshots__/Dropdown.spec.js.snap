// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Dropdown should render correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: gray;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-right: 4px;
  font-size: m;
  color: primaryText;
  text-align: left;
  line-height: normal;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: none;
  -ms-flex-wrap: none;
  flex-wrap: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  margin: 0px;
  padding: 0px;
  font-size: m;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  border: none;
  outline: none;
  background: transparent;
  color: inherit;
  cursor: pointer;
}

<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    className="c0"
    onClick={[MockFunction]}
  >
    <button
      className="c1"
      fontSize="m"
      onClick={[Function]}
      type="button"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
          color="primaryText"
          fontSize="m"
        >
          BTC
           
        </div>
        <div
          className="c4"
          color="gray"
        >
          <svg
            height="1em"
            viewBox="6.295 8.295 12 7.41"
            width="0.6em"
          >
            <path
              d="M18.295 9.705l-1.41-1.41-4.59 4.58-4.59-4.58-1.41 1.41 6 6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;
